Added information about ConcurrentModificationException (#1891)
Codecademy/docs: content/java/concepts/errors/terms/concurrentmodificationexception/concurrentmodificationexception.md

---
Title: 'ConcurrentModificationException'
Description: 'Occurs when an item is removed or added from iterable content during iteration.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Errors'
  - 'Error Handling'
  - 'Exceptions'
  - 'Debugging'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The `ConcurrentModificationException` exception is thrown when one thread is iterating through a collection using an Iterator object, and another thread modifies the collection by adding, removing, or modifying its elements. When the `Iterator` object tries to access the next element in the `collection`, it detects that the collection has been modified, and throws the `ConcurrentModificationException`.

Reviewer: ```suggestion
The `ConcurrentModificationException` exception is thrown when one thread is iterating through a collection using an `Iterator` object, and another thread modifies the collection by adding, removing, or modifying its elements. When the `Iterator` object tries to access the next element in the `collection`, it detects that the collection has been modified, and throws the `ConcurrentModificationException`.
```

> ## EXAMPLE

Reviewer 2: ```suggestion
## Example
```

Author: 🫡

> 

Reviewer: Add a brief explanation of what is happening in the example. 
```suggestion

The following example throws the `ConcurrentModificationException` because the `arr` ArrayList is modified during iteration:

```

---
Title: 'ConcurrentModificationException'

> Description: 'Occurs when item Removed or Add from Iterable content during Iteration.'

Reviewer 2: ```suggestion
Description: 'Occurs when an item is removed or added from iterable content during iteration.'
```

Author: Thanks for the suggestion.

Subjects:
  - 'Code Foundations'
  - 'Computer Science

> Tags:

Reviewer 2: Please only use tags from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).

Author: Sure.!

  - 'Errors'
  - 'Error Handling
  - 'Exceptions'
  - 'Exceptions Handling'
  - 'Iterator'
  - 'Debugging'
  - 'ConcurrentModificationException'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The `ConcurrentModificationException` exception is thrown when one thread is iterating through a collection using an Iterator object, and another thread modifies the collection by adding, removing, or modifying its elements. When the `Iterator` object tries to access the next element in the `collection`, it detects that the collection has been modified, and throws the `ConcurrentModificationException`.

> ### EXAMPLE

Reviewer 2: ```suggestion
## Example
```

```java
import java.util.ArrayList;

public class modificationError {
 public static void main(String args[])
 {
  // Creating an object of ArrayList Object
  ArrayList<String> arr = new ArrayList<String>();
  arr.add("One");
  arr.add("Two");
  arr.add("Three");
  arr.add("Four");

  try {

>    // enhanced for loop, It used the Iterator class under the hood. 

Reviewer 2: ```suggestion
   // Enhanced for loop, It uses the Iterator class under the hood. 
```

   for (String elem : arr) {
        if (elem.equals("One")) {

>             arr.remove(elem); // this will throw the exception.

Reviewer 2: ```suggestion
            arr.remove(elem); // This will throw the exception.
```
